src/hardware/siglent-sds/protocol.c

```diff
@@ -567,7 +568,7 @@ SR_PRIV int siglent_sds_receive(int fd, int revents, void *cb_data)
 				} else {
 					sr_dbg("Requesting: %" PRIu64 " bytes.", devc->num_samples - devc->num_block_bytes);
 					len = sr_scpi_read_data(scpi, (char *)devc->buffer, devc->num_samples-devc->num_block_bytes);
```

Comment:
This line here is the culprit. If I remove the subtraction devc->num_block_bytes then I get the entire waveform, no points missing.